### PR TITLE
feat: add redis dependency and configuration

### LINKED ISSUE #1779: Add Redis as a new key-value NoSQL database option
Add Redis as a key-value database option first next to MongoDB in the NoSQL options.

The idea is to add just the dependency and basic configuration to let users/developers add what they need from there.

And in another PR add the option to use it as a cache provider.

And maybe in another one, the option to use it as a message broker next to Apache Kafka.

### PR DESCRIPTION
**Pull request by frascu**
Fixes #1779

### CODE REVIEW

File: src/main/resources/generator/server/springboot/database/redis/JSR310DateConverters.java.mustache [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import org.springframework.core.convert.converter.Converter;
+
+/**
+ * <p>JSR310DateConverters class.</p>
+ */
+public final class JSR310DateConverters {
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/database/redis/JSR310DateConverters.java.mustache:11]: Can this be package private?

File: src/main/resources/generator/server/springboot/database/redis/RedisDatabaseConfiguration.java.mustache [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@EnableRedisRepositories({ "{{packageName}}" })
+@Import(value = RedisAutoConfiguration.class)
+public class RedisDatabaseConfiguration {
+
+  @Bean
+  public RedisCustomConversions customConversions() {
+    List<Converter<?, ?>> converters = new ArrayList<>();
+    converters.add(DateToZonedDateTimeConverter.INSTANCE);
+    converters.add(ZonedDateTimeToDateConverter.INSTANCE);
+    return new RedisCustomConversions(converters);
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/database/redis/RedisDatabaseConfiguration.java.mustache:25]: Maybe List.of(...) will be more readable

File: src/main/resources/generator/server/springboot/database/redis/JSR310DateConvertersTest.java.mustache [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Date;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@UnitTest
+class JSR310DateConvertersTest {
+
+  @Test
+  @DisplayName("Should convert ZonedDateTime to Date")
+  void shouldConvertZonedDateTimeToDate() {
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/database/redis/JSR310DateConvertersTest.java.mustache:17]: Can you add lines breaks to show a structure like Arrange / Act / Assert (or Given / When / Then), no need to add comments or thinks like that, just line breaks

File: src/main/resources/generator/server/springboot/database/redis/JSR310DateConvertersTest.java.mustache [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    ZonedDateTime source = ZonedDateTime.parse("2022-02-15T12:00:00+01:00[Europe/Paris]");
+    Date expected = Date.from(source.toInstant());
+    Date result = ZonedDateTimeToDateConverter.INSTANCE.convert(source);
+    assertThat(result).isEqualTo(expected);
+  }
+
+  @Test
+  @DisplayName("Should convert Date to ZonedDateTime")
+  void shouldConvertDateToZonedDateTime() {
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/database/redis/JSR310DateConvertersTest.java.mustache:26]: Can you add lines breaks to show a structure like Arrange / Act / Assert (or Given / When / Then), no need to add comments or thinks like that, just line breaks

File: src/main/resources/generator/server/springboot/database/redis/TestRedisManager.java.mustache [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package {{packageName}};
+
+import org.springframework.boot.context.event.ApplicationEnvironmentPreparedEvent;
+import org.springframework.context.ApplicationListener;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class TestRedisManager implements ApplicationListener<ApplicationEnvironmentPreparedEvent> {
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/database/redis/TestRedisManager.java.mustache:8]: Can this be package private?

File: tests-ci/start_docker_compose.sh [hunk 6]
```diff
@@ -30,6 +30,9 @@ fi
 if [ -a src/main/docker/mongodb.yml ]; then
   docker compose -f src/main/docker/mongodb.yml up -d
 fi
+if [ -a src/main/docker/redis.yml ]; then
```
Comment (DamnClin) [inline tests-ci/start_docker_compose.sh:33]: Can you also generate an app with this module in generate.sh?  We need that to test the generated application

**Review by DamnClin — COMMENTED** [review]
Thanks a lot for your first contribution! Do you have any feedback on JHlite?

**Comment by DamnClin** [comment]
Some tests are missing:

* Unit tests for the module factory;
* Component test (*.feature file).

You are also missing an `infrastructure` package before the `primary` one

**Comment by DamnClin** [comment]
Forgot to mention, this is documented here: https://github.com/jhipster/jhipster-lite/blob/main/documentation/module-creation.md

**Comment by frascu** [comment]
I fixed a typo. Now you can execute the workflow.

**Comment by DamnClin** [comment]
Looks good to me. As this is a PR with a bounty I let @pascalgrimaud merge it :)

**Comment by pascalgrimaud** [comment]
Thanks @frascu for your contribution
Thanks @DamnClin for the review

**Comment by frascu** [comment]
Bug bounty claimed 🙂 https://opencollective.com/generator-jhipster/expenses/127870

**Comment by pascalgrimaud** [comment]
@frascu : approved

**Pull request merged**